libs/libbat/src/BatteryModel.cpp

```diff
@@ -0,0 +1,157 @@
+#include <iostream>
+
+#include <verge/mission-command/BatteryModel.h>
+
+using namespace Verge::MissionCommand;
+
+BatteryModel::BatteryModel(BatteryCoefficients coefficients) : 
+    m_coefficients(coefficients),
+    m_voltage(0.0f),
+    m_current(0.0f),
+    m_capacity(0.0f),
+    m_temperature(0.0f),
+    m_rollingAverage(0.0f),
+    m_armed(false),
+    m_capacityInitialized(false) {}
+
+/**
+ * Continuous update loop of battery information
+ *
+ * @param dt interval of time elasped in seconds
+ */
+void BatteryModel::update(float dt)
+{
+    // Initialize capacity
+    if (!m_capacityInitialized)
+    {
+        m_capacityInitialized = initCapacity();
+    }
+
+    else
+    {
+        // Decrement current draw, assuming current is read at mA
+        m_capacity = m_capacity - (m_current * (dt / (3600)));
+    }
+}
+
+/**
+ * Return output of general 3rd order polynomial
+ *
+ * @param x3    3rd degree coefficient
+ * @param x2    2nd degree coefficient
+ * @param x1    1st degree coefficient
+ * @param x0    constant coefficient
+ *
+ * @return output of 3rd order polynomial
+ */
+float BatteryModel::equation(float x3, float x2, float x1, float x0, float input)
+{
+    return x3 * pow(input, 3) + x2 * pow(input, 2) + x1 * input + x0;
+}
+
+/**
+ * Uses a given voltage to estimation current capacity
+ *
+ * @param voltage battery voltage
+ * @return float capacity estimation
+ */
+float BatteryModel::voltageToCapacity(float voltage)
+{
+    return equation(m_coefficients.x3, m_coefficients.x2,
+                    m_coefficients.x1, m_coefficients.x0,
+                    voltage);
+}
+
+/**
+ * @brief Uses givne capacity and converts to time estimation
+ *
+ * @param capacity current capacity
+ * @return float capacity estimation
+ */
+float BatteryModel::capacityToTime(float capacity)
+{
+    return equation(m_coefficients.y3, m_coefficients.y2,
+                    m_coefficients.y1, m_coefficients.y0,
+                    capacity);
+}
+
+/**
+ * Initializes battery capacity based on measured voltages
+ *
+ * @return status of capacity stabilization
+ */
+bool BatteryModel::initCapacity()
+{
+    float smallestDifference = 0.2f;
```

Comment:
for values that don't change, put a `const` property in front of them. It's helpful to people reading the code that it is constant and not intended to change, and also helpful to the compiler.

```cpp
const float smallestDifference = 0.2f;
```